File: aurora-scala/src/main/scala/net/gree/aurora/scala/domain/datasource/DataSourceImpl.scala
package net.gree.aurora.scala.domain.datasource

import net.gree.aurora.domain.datasource.{DataSource => JDataSource}
import org.sisioh.dddbase.utils.{Option => JOption}

private[domain]
class DataSourceImpl(val underlying: JDataSource) extends DataSource {

  val identity = DataSourceId(underlying.getIdentity.getValue)

  val dataSourceType = underlying.getDataSourceType

  val host = underlying.getHost

  val port = {
    underlying.getPort match {
      case s: JOption.Some[_] =>
        Some(s.get.asInstanceOf[Int])
      case n: JOption.None[_] =>
        None
    }
  }

  override def toString =
    Seq(
      s"identity = $identity",
      s"host = $host",
      s"port = $port"
    ).mkString("DataSource(", ", ", ")")

}

File: aurora-core/src/main/java/net/gree/aurora/domain/datasource/DataSourceImpl.java
package net.gree.aurora.domain.datasource;

import org.sisioh.dddbase.utils.Option;

final class DataSourceImpl extends AbstractDataSource {

    DataSourceImpl(
            DataSourceId id,
            DataSourceType dataSourceType,
            String host,
            Option<Integer> port,
            Long order
    ) {
        super(id, dataSourceType, host, port, order);
    }

    @Override
    public int compareTo(DataSource o) {
        return getOrder().compareTo(o.getOrder());
    }

    @Override
    public DataSourceImpl clone() {
        return (DataSourceImpl) super.clone();
    }

    @Override
    public String toString() {
        return String.format(
                "DataSource(identity = %s, dataSourceType = %s, host = %s, port = %s)",
                getIdentity(), getDataSourceType(), getHost(), getPort()
        );
    }


}
